src/source/Ice/SocketConnection.c

```diff
@@ -329,7 +330,8 @@ STATUS socketSendDataWithRetry(PSocketConnection pSocketConnection, PBYTE buf, U
     while (socketWriteAttempt < MAX_SOCKET_WRITE_RETRY && bytesWritten < bufLen) {
         result = sendto(pSocketConnection->localSocket, buf, bufLen, NO_SIGNAL, destAddr, addrLen);
         if (result < 0) {
-            if (errno == EAGAIN || errno == EWOULDBLOCK) {
+            errorNum = errno;
```

Comment:
I believe most implementations would store this value in TLS which is pretty fast but I like this change